winforms/src/toga_winforms/widgets/base.py

```diff
@@ -16,8 +19,48 @@
 class Scalable:
     SCALE_DEFAULT_ROUNDING = ROUND_HALF_EVEN
 
-    def init_scale(self, native):
-        self.dpi_scale = native.CreateGraphics().DpiX / 96
+    def get_dpi_scale(self, native_screen):
+        screen_rect = wintypes.RECT(
+            native_screen.Bounds.Left,
+            native_screen.Bounds.Top,
+            native_screen.Bounds.Right,
+            native_screen.Bounds.Bottom,
+        )
+        windll.user32.MonitorFromRect.restype = c_void_p
+        windll.user32.MonitorFromRect.argtypes = [wintypes.RECT, wintypes.DWORD]
+        # MONITOR_DEFAULTTONEAREST = 2
+        hMonitor = windll.user32.MonitorFromRect(screen_rect, 2)
+        pScale = wintypes.UINT()
+        windll.shcore.GetScaleFactorForMonitor(c_void_p(hMonitor), byref(pScale))
+        return pScale.value / 100
+
+    @property
+    def dpi_scale(self):
+        # For Widgets which have an assigned window
+        if (
+            getattr(self, "interface", None) is not None
+            and getattr(self.interface, "window", None) is not None
+        ):
+            self._original_dpi_scale = self.interface.window._impl._original_dpi_scale
+            return self.interface.window._impl._dpi_scale
+        # For Container Widgets when not assigned to a window
+        if hasattr(self, "native_content"):
+            _dpi_scale = self.get_dpi_scale(Screen.FromControl(self.native_content))
```

Comment:
I believe this branch was the cause of "When a window is dragged from a higher scale to a lower scale screen, it does not always resize its content, menu font and toolbar font correctly." above.

I have two screens arranged one above the other. When I drag a window from the bottom screen to the top one, the code in `winforms_LocationChanged` correctly detects the moment its DPI changes, based on the rectangle of the whole window. But then when `update_scale` is called on the Window, it takes this branch, and identifies the screen based on the `native_content` rectangle – more than half of which is still on the bottom screen, unless I drag the window very fast.

Reply:
>If it's not assigned to a window, then how can it be on a screen? In fact, how can it even have a DPI? 
It shouldn't have a DPI when not on screen, but the dpi scale value was requesred by the container widget during its initialization(even when it is not assigned to a window). Hence, I had put that branch and winforms returns the primary screen when requested for a widget which is not assigned to a window. Hence, it would have used the dpi scale value of the primary screen. Looking at it now, I realize, maybe I should have modified the container to use a default scale value.